src/js/pagination.js

```diff
@@ -0,0 +1,109 @@
+
+
+import api from "./api";
+
+async function setupPagination(callback) {
+  const initialData = await getExercises(1);
+  const totalPages = initialData.totalPages;
+  let currentPage = 1;
+
+  const firstBtn = document.getElementById("first-btn");
+  const prevBtn = document.getElementById("prev-btn");
+  const nextBtn = document.getElementById("next-btn");
+  const lastBtn = document.getElementById("last-btn");
+  const pageInfo = document.getElementById("page-info");
+
+  function renderPagination() {
+    let pages = [];
+  
+    if (totalPages <= 3) {
+      // If there are 3 or fewer pages, display all pages
+      for (let i = 1; i <= totalPages; i++) {
+        pages.push(i);
+      }
+    } else {
+      // Display dots and pages based on the current page position
+      if (currentPage > 2) {
+        // Add the first page and dots if we're beyond page 2
+        pages.push(1);
+        pages.push("...");
+      }
+  
+      // Define the range to show around the current page
+      let startPage = Math.max(1, currentPage - 1);
+      let endPage = Math.min(totalPages, currentPage + 1);
+  
+      // Adjust startPage and endPage to ensure three pages are shown if possible
+      if (currentPage === 1) {
+        endPage = 3; // Show pages 1, 2, 3 at the start
+      } else if (currentPage === totalPages) {
+        startPage = totalPages - 2; // Show last three pages
+      }
+  
+      for (let i = startPage; i <= endPage; i++) {
+        pages.push(i);
+      }
+  
+      if (currentPage < totalPages - 1) {
+        // Add dots and the last page if we're not near the end
+        pages.push("...");
+        pages.push(totalPages);
+      }
+    }
+  
+    // Render pagination buttons with dots
+    pageInfo.innerHTML = pages
+      .map(page => {
+        if (page === "...") {
+          return `<span class="dots">${page}</span>`;
+        }
+        return `<button class="page-number ${page === currentPage ? 'active' : ''}" data-page="${page}">${page}</button>`;
+      })
+      .join(" ");
+  
+    callback(currentPage);
```

Comment:
the callback should be added as event listener to each button, button should have value attribute with page number, so it's reachable in the callback in event, and you don't need to call callback on each render, it should be called on button click

To test if it works, create callback, that will log to console number of the button (event.target.value)